### DISCUSSION TITLE
Running Doks on Azure Static Web Apps

### DISCUSSION THREAD

**Discussion by marie-schmidt** [thuliteio/doks#960] [Q&A]
Hi I have tried to run Hugo on an Azure Static Web App using the Doks theme. I have followed your tutorial step-by-step but ran into issues and then also tutorial available on the Microsoft documentation (https://learn.microsoft.com/en-us/azure/static-web-apps/publish-hugo). Are we missing something to be able to successfully publish this on Azure? We absolutely love your theme so we would really prefer to get this up and running! Thanks :-)
![MicrosoftTeams-image](https://user-images.githubusercontent.com/69484210/207402213-dfbbced2-237b-4afb-a5f0-90cf1c48b07e.png)

**Comment by h-enk** [comment]
Do you have a repo you could share?

On top of my head, I think there was an issue (to be solved) with Node.js 18 (v.16 should work).

**Reply by marie-schmidt** [reply]
Hi, thanks for the answer! I managed to temporarily fix the build by using this advice (see: https://github.com/microsoft/Oryx/issues/1590), nevertheless they mention that remaining in dev limits the functionality. The following repository is where I have the project: https://github.com/marie-schmidt/my-doks-site. Thanks for the feedback :-)

**Reply by h-enk** [reply]
I got it working using Node.js v14.x:

- website: https://calm-sea-09bec140f.2.azurestaticapps.net/
- repo: https://github.com/gethyas/doks-microsoft-azure
- workflow: https://github.com/gethyas/doks-microsoft-azure/blob/main/.github/workflows/azure-static-web-apps-calm-sea-09bec140f.yml
- action: https://github.com/gethyas/doks-microsoft-azure/actions/workflows/azure-static-web-apps-calm-sea-09bec140f.yml

In your project's `package json`, set:

```json
  "engines": {
    "node": "14.x"
  },
```

And (maybe not necessary, not tested using higher version): 

```json
  "otherDependencies": {
    "hugo": "0.96.0"
  }
```

Also, make sure to update `baseurl`.

P.S. I used the (very handy) [SWA Extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azurestaticwebapps) in [Visual Studio Code](https://code.visualstudio.com/).

**Reply by marie-schmidt** [reply]
Fantastic, thanks so much! Got it working by indeed setting the node version to "14.x" in my package.json file ;-) Appreciate the help on this & keep up the great work!
